feat(WA): moving calendar heatmap to the trend display

jabahamondes opened the pull request:
I recommend to review by commit because I did:
- added logic insight the trend insight
- removed old logic
- some fixes

Flag to enable: `calendar-heatmap-insight`

## Changes

Moving logic from Calendar heatmap insight into trend display type as calendar heatmap. And removing legacy logic. 

The main logic in the trend delegate the logic to the claendar heatmap query runner.

## How did you test this code?

Updated tests and manually:

https://github.com/user-attachments/assets/a496f8bf-eece-4625-9bd5-ff77c3ab96b5

With Ff disabled:
<img width="1268" height="185" alt="image" src="https://github.com/user-attachments/assets/8f08d8c8-3df1-473b-b392-4f30c30f785c" />
We just disable the entry point:
<img width="377" height="326" alt="image" src="https://github.com/user-attachments/assets/1f6819e5-c675-476f-8c34-0d907a399cb7" />


Docs PR: https://github.com/PostHog/posthog.com/pull/12542

Commit:
Big refactor where I move all the logic to the trend display. I didn't remove at this point the old code

Commit:
refactor: remove CalendarHeatmap references and update related logic

- Changed enum value from 'Calendar Heatmap' to 'CalendarHeatmap'.
- Removed CalendarHeatmapFilterType and its usages across the codebase.
- Updated related logic in ChartFilter, CommandPalette, and various query files to reflect the removal of CalendarHeatmap.
- Cleaned up unused imports and components related to CalendarHeatmap.

jabahamondes commented on frontend/src/scenes/insights/views/CalendarHeatMap/calendarHeatMapLogic.tsx:
```diff
@@ -0,0 +1,205 @@
+                            typeof result.column === 'number' &&
+                            typeof result.value === 'number'
+                        ) {
+                            // Convert ClickHouse toDayOfWeek (1=Mon, 2=Tue, ..., 7=Sun) to standard format (0=Sun, 1=Mon, ...)
+                            const standardDay = result.row % 7
+                            // Adjust for team's week start day setting
+                            const adjustedDay = (standardDay - weekStartDay + 7) % 7
```
It does because sunday is 0

Commit:
typo

Commit:
fix: add CalendarHeatmapQuery support to QueryEventsExtractor

- Add CalendarHeatmapQuery import to query_metadata.py
- Add handler for CalendarHeatmapQuery in extract_events method
- Fixes failing test_extract_events_calendar_heatmap_query test

Commit:
Merge remote-tracking branch 'origin/master' into feat/embedding-calendar-heatmap-into-trends

Commit:
fix: update TrendsSeries label condition in EditorFilters

- Adjusted the label assignment for TrendsSeries to only set it when not displaying CalendarHeatmap.

Commit:
Merge remote-tracking branch 'origin/master' into feat/embedding-calendar-heatmap-into-trends

lricoy reviewed (approved):
I think we don't need a migration logic for previously created insights, is this correct?

And if I understood correctly, this has a requirement of having a single series to make it work, I think some more guards on both frontend and backend would make sense to make this more explicit.

Personally I think keeping this separate makes more sense, but PA team will know better how to approach it.

Approving, but I think you should wait for a PA review

anirudhpillai commented on posthog/hogql_queries/insights/trends/trends_query_runner.py:
```diff
@@ -339,6 +339,58 @@ def to_actors_query_options(self) -> InsightActorsQueryOptionsResponse:
     def _calculate(self):
+        # If display type is calendar heatmap, delegate to calendar heatmap query runner
+        if self._trends_display.display_type == ChartDisplayType.CALENDAR_HEATMAP:
+            from posthog.hogql_queries.insights.trends.calendar_heatmap_query_runner import CalendarHeatmapQueryRunner
```
even better, you can move the check and the logic outside [here](https://github.com/PostHog/posthog/blob/fix/ret-cohort-actors/posthog/hogql_queries/query_runner.py#L185) so then TrendsQueryRunner doesn't need to know anything calendar specific/have circular imports

anirudhpillai commented on posthog/constants.py:
```diff
@@ -100,6 +102,7 @@ class AvailableFeature(StrEnum):
     "ActionsBarValue",
     "WorldMap",
     "BoldNumber",
+    "CalendarrHeatmap",
```
```suggestion
    "CalendarHeatmap",
```

anirudhpillai reviewed (approved):
Great change to make it part of trends!
Guess you'll update the docs to show the new flow as well, excited for users to discover and start using this :) 

From product analytics standpoint, calendar heatmap and world map as special cases of trends, maybe we can group them separately in the dropdown

Commit:
merge conflicts

Commit:
Pr Review comments: added CalendarHeatmapTrendsQueryRunner for calendar heatmap support

- Introduced CalendarHeatmapTrendsQueryRunner to handle calendar heatmap queries and return data in TrendsQueryResponse format.
- Updated TrendsQueryRunner to delegate to the new runner when the display type is CALENDAR_HEATMAP.
- Fixed typo in constants.py from CalendarrHeatmap to CalendarHeatmap.
- Added new options in ChartFilter for Space

Commit:
Renaming

Commit:
Merge remote-tracking branch 'origin/master' into feat/embedding-calendar-heatmap-into-trends

Commit:
updating schema.json file

Commit:
updating schema.py file

Commit:
Merge remote-tracking branch 'origin/master' into feat/embedding-calendar-heatmap-into-trends

Commit:
Merge remote-tracking branch 'origin/master' into feat/embedding-calendar-heatmap-into-trends

Commit:
Merge remote-tracking branch 'origin/master' into feat/embedding-calendar-heatmap-into-trends
